python/postprocessing/modules/jme/jetmetUncertainties.py

```diff
@@ -380,33 +428,52 @@ def resolution_matching(jet, genjet):
 
               # don't store the low pt jets in the Jet_pt_nom branch
               if iJet < nJet:
-                  jets_pt_jerUp    .append(jet_pt_jerUpVal*jet_pt)
-                  jets_pt_jerDown  .append(jet_pt_jerDownVal*jet_pt)
-                  jets_mass_jerUp  .append(jet_pt_jerUpVal   *jet_mass)
-                  jets_mass_jerDown.append(jet_pt_jerDownVal *jet_mass)
+                  for jerID in self.splitJERIDs:
+                      jets_pt_jerUp[jerID].append(jet_pt_jerUp[jerID])
+                      jets_pt_jerDown[jerID].append(jet_pt_jerDown[jerID])
+                      jets_mass_jerUp[jerID].append(jet_pt_jerUp[jerID])
+                      jets_mass_jerDown[jerID].append(jet_pt_jerDown[jerID])
               
               for jesUncertainty in self.jesUncertainties:
                   # (cf. https://twiki.cern.ch/twiki/bin/view/CMSPublic/WorkBookJetEnergyCorrections#JetCorUncertainties )
-                  self.jesUncertainty[jesUncertainty].setJetPt(jet_pt_nom)
-                  self.jesUncertainty[jesUncertainty].setJetEta(jet.eta)
-                  delta = self.jesUncertainty[jesUncertainty].getUncertainty(True)
-                  jet_pt_jesUp[jesUncertainty]   = jet_pt_nom*(1. + delta)
-                  jet_pt_jesDown[jesUncertainty] = jet_pt_nom*(1. - delta)
+
+                  # cf. https://hypernews.cern.ch/HyperNews/CMS/get/JetMET/2000.html
+                  if jesUncertainty == "HEMIssue":
+                      delta = 1.
+                      if jet.phi > -1.57 and jet.phi < -0.87:
+                          if jet.eta > -2.5 and jet.eta < -1.3:
+                            delta = 0.8
+                          elif jet.eta <= -2.5 and jet.eta > -3:
+                            delta = 0.65
+                      jet_pt_jesUp[jesUncertainty] = jet_pt_nom
+                      jet_pt_jesDown[jesUncertainty] = delta * jet_pt_nom
+                      jet_mass_jesUp[jesUncertainty] = jet_mass_nom
```

Comment:
Should the delta also be applied to the jet mass? This is done for "regular" JES in the code below

Reply:
Ah, never mind, jes up == nominal here. I see it is applied for the jes down.

Reply:
Right, for the HEM issue the variation is single-sided.